litellm/integrations/anycost/utils.py

```diff
@@ -0,0 +1,627 @@
+
+            # Create group key
+            group_key = (team_id, service_name, provider, model)
+
+            if group_key not in grouped:
+                grouped[group_key] = []
+            grouped[group_key].append(record)
+
+        return grouped
+
+    def _calculate_group_metrics(self, records: List[Any]) -> Dict[str, Any]:
+        """Calculate aggregated metrics for a group of records"""
+        metrics = {
+            'spend': 0.0,
+            'prompt_tokens': 0,
+            'completion_tokens': 0,
+            'cache_read_input_tokens': 0,
+            'cache_creation_input_tokens': 0,
+            'api_requests': 0,
+            'successful_requests': 0,
+            'failed_requests': 0,
+        }
+
+        for record in records:
+            metrics['spend'] += getattr(record, 'spend', 0)
+            metrics['prompt_tokens'] += getattr(record, 'prompt_tokens', 0)
+            metrics['completion_tokens'] += getattr(record, 'completion_tokens', 0)
+            metrics['cache_read_input_tokens'] += getattr(
+                record, 'cache_read_input_tokens', 0
+            )
+            metrics['cache_creation_input_tokens'] += getattr(
+                record, 'cache_creation_input_tokens', 0
+            )
+            metrics['api_requests'] += getattr(record, 'api_requests', 0)
+            metrics['successful_requests'] += getattr(record, 'successful_requests', 0)
+            metrics['failed_requests'] += getattr(record, 'failed_requests', 0)
+
+        metrics['total_tokens'] = metrics['prompt_tokens'] + metrics['completion_tokens']
+        return metrics
+
+    def _create_cbf_line_item(
+        self,
+        team_id: str,
+        service_name: str,
+        provider: str,
+        model: str,
+        metrics: Dict[str, Any],
+        billing_period: CBFBillingPeriod
+    ) -> CBFLineItem:
+        """Create a CBF line item for a specific breakdown"""
+        usage_info = [
+            CBFUsageInfo(
+                metric_name="API Requests",
+                usage_amount=float(metrics['api_requests']),
+                usage_unit="requests"
+            ),
+            CBFUsageInfo(
+                metric_name="Total Tokens",
+                usage_amount=float(metrics['total_tokens']),
+                usage_unit="tokens"
+            ),
+            CBFUsageInfo(
+                metric_name="Prompt Tokens",
+                usage_amount=float(metrics['prompt_tokens']),
+                usage_unit="tokens"
+            ),
+            CBFUsageInfo(
+                metric_name="Completion Tokens",
+                usage_amount=float(metrics['completion_tokens']),
+                usage_unit="tokens"
+            )
+        ]
+
+        # Create unique line item ID
+        date_str = billing_period.start_date[:10]
+        line_item_id = f"litellm-{team_id}-{service_name[:16]}-{provider}-{model}-{date_str}".replace(' ', '-').replace('(', '').replace(')', '')
```

Comment:
key_alias is a default, and then we can override with key metadata 

proposal - allow defining a `line_item_id` for a key (the goal here is to add billing metadata). billing_tags=["team1, team2"]